src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -828,6 +879,185 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
             }
         }
 
+        public async Task WaitForDatabaseNotificationOnCluster(string databaseName, long index,
+            Dictionary<string, string> clusterNodes, // (nodeTag, url)
+            HashSet<string> databaseNodes, // nodeTags
+            CancellationToken token)
+        {
+            try
+            {
+
+                foreach (var (tag, nodeUrl) in clusterNodes)
+                {
+                    if (tag == ServerStore.NodeTag)
+                    {
+                        await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, index);
+
+                        if (databaseNodes.Contains(tag))
+                        {
+                            try
+                            {
+                                var database = await ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName);
```

Comment:
I want a single code path here.... so let the node contact itself via the same EP

Reply:
Done